src/modm/platform/clock/sam_pmc/clockgen.cpp.in

```diff
@@ -81,26 +86,41 @@ uint32_t ClockGen::mainClkFrequency() {
 	// a way for user supplied input frequency. For now, this assumes we're
 	// using the internal RC for main clock.
 	uint32_t moscrcf = (PMC->CKGR_MOR & CKGR_MOR_MOSCRCF_Msk) >> CKGR_MOR_MOSCRCF_Pos;
-	if(moscrcf == (uint32_t)MainInternalFreq::Rc8Mhz) {
-		return 8'000'000;
-	} else if(moscrcf == (uint32_t)MainInternalFreq::Rc16Mhz) {
-		return 16'000'000;
-	} else if(moscrcf == (uint32_t)MainInternalFreq::Rc24Mhz) {
-		return 24'000'000;
-	} else {
-		return 0;
-	}
+	const auto freq = MainInternalFreq(moscrcf);
+#ifdef CKGR_MOR_MOSCRCF_4_MHz_Val
+	if (freq == MainInternalFreq::Rc4Mhz) return 4'000'000;
+#endif
+#ifdef CKGR_MOR_MOSCRCF_8_MHz_Val
+	if (freq == MainInternalFreq::Rc8Mhz) return 8'000'000;
+#endif
+#ifdef CKGR_MOR_MOSCRCF_12_MHz_Val
+	if (freq == MainInternalFreq::Rc12Mhz) return 12'000'000;
+#endif
+#ifdef CKGR_MOR_MOSCRCF_16_MHz_Val
+	if (freq == MainInternalFreq::Rc16Mhz) return 16'000'000;
+#endif
+#ifdef CKGR_MOR_MOSCRCF_24_MHz_Val
+	if (freq == MainInternalFreq::Rc24Mhz) return 24'000'000;
+#endif
+	return 0;
 }
 
 uint32_t ClockGen::pllAFrequency() {
 	uint32_t mul = ((PMC->CKGR_PLLAR & CKGR_PLLAR_MULA_Msk) >> CKGR_PLLAR_MULA_Pos) + 1;
+%% if target["family"] in ["g"]
 	uint32_t freq = SlowClkFreqHz * mul;
 	if(PMC->PMC_MCKR & PMC_MCKR_PLLADIV2) {
 		freq /= 2;
 	}
+%% elif target["family"] in ["v"]
+	uint32_t freq = mainClkFrequency() * mul;
+	// division by zero error if PLLA is not enabled
```

Comment:
👀 wanna fix that? Could be painful to debug.

```cpp
const uint32_t div = ((PMC->CKGR_PLLAR & CKGR_PLLAR_DIVA_Msk) >> CKGR_PLLAR_DIVA_Pos);
if (div) freq /= div;
else freq = freq; // or whatever the right math is?
```

Reply:
This was rather a loud exception-like error and not an omission. But sure, graceful return won't hurt.